src/textual/binding.py

```diff
@@ -252,3 +360,61 @@ def get_bindings_for_key(self, key: str) -> list[Binding]:
             return self.key_to_bindings[key]
         except KeyError:
             raise NoBinding(f"No binding for {key}") from None
+
+
+BindingIDString = str
+"""The ID of a Binding defined somewhere in the application.
+
+Corresponds to the `id` parameter of the `Binding` class.
+"""
+
+KeyString = str
+"""A string that represents a key binding.
+
+For example, "x", "ctrl+i", "ctrl+shift+a", "ctrl+j,space,x", etc.
+"""
+
+
+class Keymap:
+    """A mapping of binding IDs to key strings.
+
+    Used by Textual on startup to override default key bindings.
+
+    Override App.get_keymap() in an App subclass to have Textual
+    replace bindings defined across that App with the overrides defined
+    in the keymap.
+
+    Textual will look for bindings with the same ID as the
+    the keys in this mapping, and replace the key strings in the
+    default binding with the key strings from this mapping.
+
+    The mapping is immutable - it's created once and then should
+    never be mutated at runtime.
+    """
+
+    def __init__(self, mapping: Mapping[BindingIDString, KeyString]):
```

Comment:
Maybe make this positional only?

Reply:
I can't even remember how to do this 😂 not really sold on them having any benefits though (open for convincing if you care enough).

Reply:
You add a forward slash. But you're right, not much benefit now I think about it.